lib/AnalysisStructured/PtrAnalysis.cpp

```diff
@@ -1026,8 +1026,11 @@ LogicalResult PtrAnalysis::rewriteAddptrOp(triton::AddPtrOp op) {
       // continuous dimensions.
       if (state.getRank() == 1)
         return failure();
-      auto maketptrOp = state.createTTSMakeGatherScatterTensorPtrOp(builder, op.getLoc());
-      ptrMap.map(op.getResult(), maketptrOp.getResult());
+      if (enableMakeGatherScatterTensorPtr) {
+        auto maketptrOp =
+            state.createTTSMakeGatherScatterTensorPtrOp(builder, op.getLoc());
+        ptrMap.map(op.getResult(), maketptrOp.getResult());
+      }
     }
```

Comment:
i think this whole block should be wrapped inside if enableMakeGatherScatterTensorPtr. we should just return failure() if enableMakeGatherScatterTensorPtr is false.

Reply:
Fixed.